clients/common/blis_interface.cpp

```diff
@@ -0,0 +1,95 @@
+#include "blis_interface.hpp"
+#include "blis.h"
+#include "omp.h"
+
+trans_t blis_transpose(rocblas_operation trans)
+{
+    if(trans == rocblas_operation_none)
+    {
+        return BLIS_CONJ_NO_TRANSPOSE;
+    }
+    else if(trans == rocblas_operation_transpose)
+    {
+        return BLIS_CONJ_TRANSPOSE;
+    }
+    else if(trans == rocblas_operation_conjugate_transpose)
+    {
+        return BLIS_CONJ_TRANSPOSE;
+    }
+    else
+    {
+        std::cerr << "rocblas ERROR: trans != N, T, C" << std::endl;
+        exit(1);
+    }
+}
+
+void setup_blis()
+{
+    bli_init();
+    bli_thread_set_num_threads(omp_get_max_threads());
+}

```

Comment:
Rather than explicitly call `setup_blis` from every client, have a static variable which initializes it, e.g.:

```
static int initialize_blis = (setup_blis(), 0);
```
And put that in `blis_interface.cpp`.

If calling BLIS before `main()` starts is required, I can help provide solutions using function-local static variables.